cypress/e2e/buy-crypto-modal.test.ts

```diff
@@ -0,0 +1,36 @@
+import { getTestSelector } from '../utils'
+
+describe('Buy Crypto Modal', () => {
+  it('should open and close', () => {
+    cy.visit('/')
+
+    cy.get(getTestSelector('buy-fiat-button'))
+      .click()
+      .then(() => {
+        cy.get(getTestSelector('fiat-onramp-modal')).should('be.visible')
+      })
+
+    cy.get('body')
```

Comment:
Add comments (preferably before every block except `cy.visit` in your test) explaining what they're doing. eg

    // Click outside the modal to dismiss it.